server/src/test/java/org/elasticsearch/index/translog/TranslogTests.java

```diff
@@ -1507,6 +1510,140 @@ public void testSnapshotFromStreamInput() throws IOException {
         assertEquals(ops, readOperations);
     }
 
+    public void testSnapshotTrimmedOperations() throws Exception {
+        int translogOperations = 0;
+        int skipped = 0;
+        Set<Long> expectedSeqNo = new HashSet<>();
+
+        for(int attempt = 0, maxAttempts = randomIntBetween(3, 10); attempt < maxAttempts; attempt++) {
+            int extraDocs = randomIntBetween(10, 15);
+
+            List<Translog.Index> operations = new ArrayList<>();
+            for (int op = 0; op < extraDocs; op++) {
+                String ascii = randomAlphaOfLengthBetween(1, 50);
+                int seqNo = translogOperations + op;
+                Translog.Index operation = new Translog.Index("test", "" + seqNo, seqNo,
+                    this.primaryTerm.get(), ascii.getBytes("UTF-8"));
+                operations.add(operation);
+            }
+            Randomness.shuffle(operations);
+
+            for (Translog.Index operation : operations) {
+                translog.add(operation);
+            }
+
+            final boolean rollover = randomBoolean();
+            if (rollover) {
```

Comment:
can you please inline this while [adding docs](https://github.com/elastic/elasticsearch/pull/30176/files#diff-ed6e20d0c4d03d97ae9b7a9a33190c4bR1532)? We need to have more roll overs randomly

Reply:
While you do the rollover, please increment the translog term too.

Reply:
it is in a loop.